source/keyboardHandler.py

```diff
@@ -136,20 +133,30 @@ def getNVDAModifierKeys() -> List[Tuple[int, Optional[bool]]]:
 	return keys
 
 
-def shouldUseToUnicodeEx(focus=None):
+def shouldUseToUnicodeEx(focus: Optional["NVDAObject"] = None):
 	"Returns whether to use ToUnicodeEx to determine typed characters."
 	if not focus:
 		focus = api.getFocusObject()
+	from NVDAObjects.window import Window
 	from NVDAObjects.behaviors import KeyboardHandlerBasedTypedCharSupport
 	return (
 		# This is only possible in Windows 10 1607 and above
 		winVersion.getWinVer() >= winVersion.WIN10_1607
 		and (  # Either of
 			# We couldn't inject in-process, and its not a legacy console window without keyboard support.
 			# console windows have their own specific typed character support.
-			(not focus.appModule.helperLocalBindingHandle and focus.windowClassName != 'ConsoleWindowClass')
+			(
+				not focus.appModule.helperLocalBindingHandle
+				and (
+					not isinstance(focus, Window)
+					or focus.windowClassName != 'ConsoleWindowClass'
+				)
+			)
 			# or the focus is within a UWP app, where WM_CHAR never gets sent
-			or focus.windowClassName.startswith('Windows.UI.Core')
+			or (
+				not isinstance(focus, Window)

```

Comment:
I think this one is wrong.

```suggestion
				isinstance(focus, Window)
```